[J4Joshua] iP

J4Joshua opened the pull request:
# Ekko Chatbot Application

Ekko keeps track of your tasks. 🎉 :

Forked from [Github/CS2103T](https://github.com/nus-cs2103-AY2425S2/ip)

## Features
- Task management
- Command parsing
- User-friendly interface
- Persistent storage
- Error handling

## Key Components
1. Ekko Class - The main application entry point.
2. Storage Class - Handles data persistence.
3. Task Package - Manages the task classes and TaskList class
4. Ui Class - Manages user interactions.
5. Parser Class - Interprets user commands.
6. Command Package - Manages command classes
> [!TIP]
> See *command Package* for list of possible commands.

Here's the main method:
```java
public static void main(String[] args) {
        new Ekko("data/ekko.txt").run();
    }
```

Todo: 
- [x] Apply GMFD
- [ ] A-CheckStyle
- [ ] Level-10
- [ ] A-Varargs

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level 0. Rename, Greet, Exit

Commit:
Level 2. Add, List

Commit:
Level 1. Echo

Commit:
Level 2. Add, List

Commit:
Level 3. Mark as Done

Commit:
Level 4. ToDos, Events, Deadlines

Commit:
Automated Text UI Testing

Commit:
Level 5. Handle Errors

Commit:
Level 6. Delete

Commit:
Use Enums

Commit:
Level 7. Save

Commit:
Merge Level 7 implementation

Commit:
A-MoreOOP

Commit:
A-Packages

Commit:
Merge remote-tracking branch 'upstream/add-gradle-support'
merge add-gradle-support

Commit:
add dates

Commit:
add more OOP (done before level 8)

Commit:
add JUnit tests

Commit:
clean

Commit:
add javadoc

Commit:
fix coding standard

Commit:
merge A-CodingStandard

Commit:
add find and add parser and add command

Commit:
resolved conflicts

Commit:
Fix coding standard

Commit:
Add find function

zedonggg commented on src/main/java/Ekko.java:
```diff
@@ -0,0 +1,49 @@
+
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
```
Perhaps it would be more descriptive to name the boolean isExitCommand?

zedonggg commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+import ui.Ui;
+import storage.Storage;
+import exception.EkkoException;
+
```
Perhaps it would be neater to organize the import order alphabetically

zedonggg commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Represents an executable command.
+ */
+public abstract class Command {
```
I like the use of OOP, especially for all the commands 👍

zedonggg commented on src/main/java/command/DeleteCommand.java:
```diff
@@ -0,0 +1,37 @@
+        ui.showLine();
+        ui.showMessage("Noted. I've removed this task:");
+        ui.showMessage("[" + removedTask.getCategory() + "][" + removedTask.getStatusIcon() + "] " + removedTask);
+        ui.showMessage("Now you have " + tasks.size() + " tasks in the list.");
```
Would it be better to combine it as one String? That way, you only need to call showMessage once

zedonggg commented on src/main/java/parser/Parser.java:
```diff
@@ -0,0 +1,53 @@
+        String commandWord = inputArr[0];
+
+        switch (commandWord) {
+            case "list":
```
Double check just in case but I believe the course style guide recommends the case statements to be inline with the switch statement

zedonggg commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,102 @@
+import task.Task;
+import task.ToDo;
+
+import java.io.*;
```
Would it be good for the specific libraries to be specified rather than a wildcard import?

zedonggg reviewed (commented):
LGTM. Love the good usage of OOP. Your code is clear and very pleasant to read. Just some minor nits here and there. 👍

Commit:
fix coding style

Commit:
add GUI

Commit:
add assertions

J4Joshua committed:
Merge pull request #1 from J4Joshua/branch-A-Assertions

Add Assertions

- Standardized assertion checks to document important assumptions across the codebase.
- Ensured input validation with assertions for key methods in `Parser`, `TaskList`.
- Enabled better debugging and early error detection with `assert` statements.

This refactor improves code readability, reduces redundancy, and prepares for further unification of similar methods.

Commit:
improve parseDateTime

J4Joshua committed:
Merge pull request #2 from J4Joshua/branch-A-CodeQuality

Remove Arrowhead Anti-Pattern in parseDateTime()

- Refactored `parseDateTime()` to eliminate deeply nested try-catch blocks.
- Improved readability by handling parsing attempts separately.
- Added an assertion to ensure `dateTimeString` is never null.
- Enhanced maintainability by making the error-handling flow clearer.

This refactor makes the code more readable and easier to debug while maintaining functionality.

Commit:
feat: Transition from text-based UI to GUI interface

Refactored the chatbot's UI from a text-based console interface to a
graphical user interface (GUI) using JavaFX.

- Replaced `Ui` class with `MainWindow` and `DialogBox` for GUI interactions.
- Updated `Ekko` to return responses instead of printing to console.
- Modified command execution to return strings for GUI display.
- Adjusted `AddCommand`, `FindCommand`, `MarkCommand`, and `UnmarkCommand`
  to interact with the GUI.
- Removed obsolete text-based UI methods such as `showMessage()`,
  `showLine()`, and `showError()`.

This transition improves user experience by providing a more intuitive and
interactive interface.

Commit:
Add User Guide

Commit:
Add Ui.png
